api/v1alpha1/selfnoderemediationconfig_types.go

```diff
@@ -25,102 +27,109 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 const (
-	ConfigCRName                         = "self-node-remediation-config"
-	defaultWatchdogPath                  = "/dev/watchdog"
-	DefaultSafeToAssumeNodeRebootTimeout = 180
-	defaultIsSoftwareRebootEnabled       = true
+	ConfigCRName        = "self-node-remediation-config"
+	defaultWatchdogPath = "/dev/watchdog"
+	// FallbackRebootDuration is used by the manager in case it fails to calculate the minimum safe reboot time
+	FallbackRebootDuration         = 300 * time.Second
+	defaultIsSoftwareRebootEnabled = true
 )
 
 // SelfNodeRemediationConfigSpec defines the desired state of SelfNodeRemediationConfig
 type SelfNodeRemediationConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// WatchdogFilePath is the watchdog file path that should be available on each node, e.g. /dev/watchdog
+	// WatchdogFilePath is the watchdog file path that should be available on each node, e.g. /dev/watchdog.
 	// +kubebuilder:default=/dev/watchdog
+	// +optional
 	WatchdogFilePath string `json:"watchdogFilePath,omitempty"`
 
 	// SafeTimeToAssumeNodeRebootedSeconds is the time after which the healthy self node remediation
 	// agents will assume the unhealthy node has been rebooted, and it is safe to recover affected workloads.
 	// This is extremely important as starting replacement Pods while they are still running on the failed
 	// node will likely lead to data corruption and violation of run-once semantics.
 	// In an effort to prevent this, the operator ignores values lower than a minimum calculated from the
-	// ApiCheckInterval, ApiServerTimeout, MaxApiErrorThreshold, PeerDialTimeout, and PeerRequestTimeout fields.
-	// +kubebuilder:validation:Minimum=0
-	// +kubebuilder:default=180
-	SafeTimeToAssumeNodeRebootedSeconds int `json:"safeTimeToAssumeNodeRebootedSeconds,omitempty"`
+	// ApiCheckInterval, ApiServerTimeout, MaxApiErrorThreshold, PeerDialTimeout, and PeerRequestTimeout fields,
+	// and the unhealthy node's individual watchdog timeout.
+	// +optional
+	SafeTimeToAssumeNodeRebootedSeconds *int `json:"safeTimeToAssumeNodeRebootedSeconds,omitempty"`
 
+	// The timeout for api-server connectivity check.
 	// Valid time units are "ms", "s", "m", "h".
-	// +optional
 	// +kubebuilder:default:="5s"
-	// +kubebuilder:validation:Pattern="^(0|([0-9]+(\\.[0-9]+)?(ms|s|m|h)))$"
+	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
```

Comment:
well..... since the very small units are technically allowed for duration, in practice there is no use case to use them IMHO. Documenting them might be more confusing than helpful. Having `ms` in the doc is strange already IMHO. WDYT?